File: Python/Warmup-1/monkey_trouble.py
import unittest

'''

We have two monkeys, a and b, and the parameters a_smile and b_smile indicate if each is smiling. 
We are in trouble if they are both smiling or if neither of them is smiling. Return True if we are in trouble.


monkey_trouble(True, True)     -> True
monkey_trouble(False, False)   -> True
monkey_trouble(True, False)    -> False

'''

def monkey_trouble(a_smile, b_smile):
  if a_smile and not b_smile:
    return False
  elif not a_smile and b_smile:
   return False
  else:
    return True  


class Test_monkey_trouble(unittest.TestCase):

    def test_1(self):
        self.assertEqual(monkey_trouble(True, True),   True)

    def test_2(self):
        self.assertEqual(monkey_trouble(False, False), True)

    def test_3(self):
        self.assertEqual(monkey_trouble(True, False),  False)


if __name__ == '__main__':
    unittest.main() 

File: Java/Warmup_1/monkeyTrouble.java
/**
 * @author Luis Teixeira
 *
 * @date   28/03/2018
 *
 * @mail   filipe.teixeira.996@gmail.com
 */
package Warmup_1;

import static org.junit.Assert.*;

import org.junit.Test;

/*
We have two monkeys, a and b, and the parameters aSmile and bSmile 
indicate if each is smiling. We are in trouble if they are both 
smiling or if neither of them is smiling. Return true if we are in trouble.


monkeyTrouble(true, true) --> true
monkeyTrouble(false, false) --> true
monkeyTrouble(true, false) --> false
 */
public class monkeyTrouble {

	private static int i = 1;
	
	@Test
	public void Test1()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("monkeyTrouble(true, true) --> true\n");
		assertEquals("monkeyTrouble(true, true) --> true", true, monkeyTrouble(true, true));
		i++;
	}
	
	@Test
	public void Test2()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("monkeyTrouble(false, false) --> true\n");
		assertEquals("monkeyTrouble(false, false) --> true", true, monkeyTrouble(false, false));
		i++;
	}
	
	@Test
	public void Test3()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("monkeyTrouble(true, false) --> false\n");
		assertEquals("monkeyTrouble(true, false) --> false", false, monkeyTrouble(true, false));
		i++;
	}

	
	public boolean monkeyTrouble(boolean aSmile, boolean bSmile) {
		
		if ((aSmile==true && bSmile==true)||(aSmile==false && bSmile==false)) {
			return true;
		}
		
		else {
			return false;
		}
		
	}

}

